tests/example_http_conn_state.rs

```diff
@@ -0,0 +1,41 @@
+mod test_server;
+
+use rama::{error::BoxError, http::Request};
+
+use crate::test_server::recive_as_string;
+
+#[tokio::test]
+async fn test_http_conn_state() -> Result<(), BoxError> {
+    let _example = test_server::run_example_server("http_conn_state");
+
+    let get_request = Request::builder()
+        .method("GET")
+        .uri("http://127.0.0.1:40000/")
+        .body(String::new())
+        .unwrap();
+
+    let (_, res_str) = recive_as_string(get_request).await?;
+    let head = r##"
```

Comment:
As I said in previous comment, these tests are too detailed. Just testing a small port of it (e.g. contains `Connection ...` is good enough).

Reply:
In my environment, the test returns 1 or 2 after `Connection`. I am attempting to check only that it contains the string `Connection`.

Reply:
That is good enough for this PR, agreed. Could you put a comment though with “TODO: Investigate why this counter is not consistent 1”